src/frontend/sider/Fagsak/Behandling/Sider/Vedtak/Vedtaksperioder/VedtaksperiodeContext.tsx

```diff
@@ -0,0 +1,319 @@
+import { grupperBegrunnelser } from './utils';
+
+interface IProps extends PropsWithChildren {
+    vedtaksperiodeMedBegrunnelser: IVedtaksperiodeMedBegrunnelser;
+    åpenBehandling: IBehandling;
+}
+
+interface BegrunnelserSkjema {
+    periode: IIsoDatoPeriode;
+    fritekster: FeltState<IFritekstFelt>[];
+}
+
+interface VedtaksperiodeContextValue {
+    erPanelEkspandert: boolean;
+    grupperteBegrunnelser: GroupBase<OptionType>[];
+    id: number;
+    hentFeilTilOppsummering: () => FeiloppsummeringFeil[];
+    leggTilFritekst: () => void;
+    maksAntallKulepunkter: number;
+    makslengdeFritekst: number;
+    onChangeBegrunnelse: (action: ActionMeta<OptionType>) => void;
+    onPanelClose: (visAlert: boolean) => void;
+    putVedtaksperiodeMedFritekster: () => void;
+    skjema: ISkjema<BegrunnelserSkjema, IVedtaksperiodeMedBegrunnelser[]>;
+    vedtaksperiodeMedBegrunnelser: IVedtaksperiodeMedBegrunnelser;
+    åpenBehandling: IBehandling;
+    standardBegrunnelserPut: Ressurs<string>;
+    genererteBrevbegrunnelser: Ressurs<string[]>;
+}
+
+const VedtaksperiodeContext = createContext<VedtaksperiodeContextValue | undefined>(undefined);
+
+export const VedtaksperiodeProvider = ({
+    åpenBehandling,
+    vedtaksperiodeMedBegrunnelser,
+    children,
+}: IProps) => {
+    const { request } = useHttp();
+    const { vedtaksbegrunnelseTekster } = useVedtakContext();
+    const { settVedtaksperioderMedBegrunnelserRessurs } = useVedtakContext();
+
+    const [erPanelEkspandert, settErPanelEkspandert] = useState(
+        åpenBehandling.type === Behandlingstype.FØRSTEGANGSBEHANDLING &&
+            vedtaksperiodeMedBegrunnelser.begrunnelser.length === 0 &&
+            vedtaksperiodeMedBegrunnelser.fritekster.length === 0
+    );
+    const [standardBegrunnelserPut, settStandardBegrunnelserPut] =
+        useState<Ressurs<string>>(byggTomRessurs());
+    const [genererteBrevbegrunnelser, settGenererteBrevbegrunnelser] =
+        useState<Ressurs<string[]>>(byggTomRessurs());
+
+    const maksAntallKulepunkter = 3;
+    const makslengdeFritekst = 350;
+
+    const periode = useFelt<IIsoDatoPeriode>({
+        verdi: {
+            fom: vedtaksperiodeMedBegrunnelser.fom,
+            tom: vedtaksperiodeMedBegrunnelser.tom,
+        },
+    });
+
+    const fritekster = useFelt<FeltState<IFritekstFelt>[]>({
+        verdi: [],
+        valideringsfunksjon: (felt: FeltState<FeltState<IFritekstFelt>[]>) => {
+            return felt.verdi.some(
+                fritekst =>
+                    fritekst.valideringsstatus === Valideringsstatus.FEIL ||
+                    fritekst.verdi.tekst.length === 0
+            )
+                ? feil(felt, '')
+                : ok(felt);
+        },
+    });
+
+    const { hentFeilTilOppsummering, kanSendeSkjema, settVisfeilmeldinger, skjema } = useSkjema<
+        BegrunnelserSkjema,
+        IVedtaksperiodeMedBegrunnelser[]
+    >({
+        felter: {
+            periode,
+            fritekster,
```

Comment:
Ser ikke ut som periode-feltet blir brukt til noe. Her kunne vi kanskje ha skrevet oss om til at fritekster bare er en state og ikke bruke skjema-hook? Det blir for stor omskriving for denne PRen, men kan være relevant for noen andre å se på senere?